src/profiles.js

```diff
@@ -0,0 +1,199 @@
+import tv4 from 'tv4'
+import jsonlint from 'json-lint'
+import _ from 'lodash'
+import path from 'path'
+
+import Utils from './utils'
+
+const DEFAULT_REMOTE_PATH = 'http://schemas.datapackages.org/registry.csv'
+let DEFAULT_LOCAL_PATH
+if (!Utils.isBrowser) {
+  DEFAULT_LOCAL_PATH = path.join(__dirname, '..', 'schemas', 'registry.csv')
+}
+
+class Profiles {
+
+  constructor(remote = false) {
+    const self = this
+        , PATH = remote || Utils.isBrowser ? DEFAULT_REMOTE_PATH :
+                 DEFAULT_LOCAL_PATH
+
+    return new Promise((resolve, reject) => {
+      const promises = []
+      promises.push(this._loadRegistry(PATH))
+      promises.push(this._loadBasePath(PATH))
+
+      Promise.all(promises).then(values => {
+        self._registry = values[0]
+        self._base_path = values[1]
+        this._getProfiles().then(profiles => {
+          self._allProfiles = profiles
+          resolve(self)
+        }).catch(err => {
+          reject(err)
+        })
+      }).catch(err => {
+        reject(err)
+      })
+    })
+  }
+
+  /**
+   * Retrieve a profile by id.
+   *
+   * @param {String} [profile='base']
+   * @return {Promise}
+   */
+  retrieve(profile = 'base') {
+    return this._allProfiles[profile]
+  }
+
+  /**
+   * Validate a descriptor against a profile. You can provide custom schema
+   * or specify the profile id.
+   *
+   * @param {String|JSON} descriptor The descriptor that needs to be validated
```

Comment:
We don't need to support descriptor as a string only object. Also you use `JSON` to say object is expected. Not sure it's correct (not sure of otherwise).